### PR TITLE
add CloneSetHashOnlyRevisionName feature-gate

### PR DESCRIPTION
**Pull request by fatedier**
### Ⅰ. Describe what this PR does
Add ClonetSetSimpleRevisionLabel featrue-gate, default value is false. If enabled, CloneSet will create pod with simple revision hash label not including it's name.

So we can create CloneSet with name length greater than 63.

### Ⅱ. Does this pull request fix one issue?

Fixes https://github.com/openkruise/kruise/issues/531

### CODE REVIEW

File: pkg/features/kruise_features.go [hunk 1]
```diff
@@ -32,11 +32,15 @@ const (
 
 	// PodWebhook enables webhook for Pods creations. This is also related to SidecarSet.
 	PodWebhook featuregate.Feature = "PodWebhook"
+
+	// ClonetSetSimpleRevisionLabel enables simple revision hash label for CloneSet.
+	ClonetSetSimpleRevisionLabel featuregate.Feature = "ClonetSetSimpleRevisionLabel"
```
Comment (Fei-Guo) [inline pkg/features/kruise_features.go:37]: How about change the feature name to "ClonetSetHashOnlyRevisionName" ?

**Review by Fei-Guo — COMMENTED** [review]
This looks reasonable to me. A minor comment below.

Can you exclude all the changes to the vendor directory and go.sum? The PR should only include the necessary changes for the feature.

**Comment by fatedier** [comment]
> Can you exclude all the changes to the vendor directory and go.sum? The PR should only include the necessary changes for the feature.

image


@Fei-Guo I remove the changes of vendor files and it failed on unit-tests. Vendor files are auto generated after i execute `make generate manifests`.

**Comment by Fei-Guo** [comment]
> > Can you exclude all the changes to the vendor directory and go.sum? The PR should only include the necessary changes for the feature.


Can you please double check if this import "k8s.io/kubernetes/pkg/controller/history" introduces so many vendor code changes? If so,  we may just redefine the `history.ControllerRevisionHashLabel` in cloneset repo to avoid Kubernetes dependency.

**Comment by fatedier** [comment]
> > > Can you exclude all the changes to the vendor directory and go.sum? The PR should only include the necessary changes for the feature.
> 
> Can you please double check if this import "k8s.io/kubernetes/pkg/controller/history" introduces so many vendor code changes? If so, we may just redefine the `history.ControllerRevisionHashLabel` in cloneset repo to avoid Kubernetes dependency.

@Fei-Guo It's strange that `go.mod and go.sum` are auto modified when i run `go mod tidy` in my local environment. I remove the changes manually and it works now.

**Pull request merged**